packages/node/src/indexer/api.service.ts

```diff
@@ -205,4 +419,98 @@ export class ApiService implements OnApplicationShutdown {
 
     return `api.rpc.${ext?.section ?? '*'}.${ext?.method ?? '*'}`;
   }
+
+  private async fetchBlocksFromFirstAvailableEndpoint(
+    batch: number[],
+    overallSpecVer?: number,
+  ): Promise<BlockContent[]> {
+    let reconnectAttempts = 0;
+    while (reconnectAttempts < MAX_RECONNECT_ATTEMPTS) {
+      try {
+        const start = performance.now();
+        const index = this.getFirstConnectedApiIndex();
+        if (index === -1) {
+          throw new Error('No connected api');
+        }
+        const blocks = await this.fetchBlocksBatches(
+          this.getApi(index),
+          batch,
+          overallSpecVer,
+        );
+        const end = performance.now();
+        this.loadBalancer.addToBuffer(index, (end - start) / batch.length);
+        return blocks;
+      } catch (e) {
+        logger.error(e, 'Failed to fetch blocks');
+        reconnectAttempts++;
+      }
+    }
+    throw new Error(
+      `Maximum number of retries (${MAX_RECONNECT_ATTEMPTS}) reached.`,
+    );
+  }
+
+  async fetchBlocks(
```

Comment:
Lets remove all the weights and connection logic here. This function should do 3 things. 

1. Get an api instance that is connected
2. Call `fetchBlocksBatches`
3. If it fails recurse up to N times

Having all this extra logic doesn't make sense to only apply to fetching block when there are other api requests as well.